prov/sm2/src/sm2_rma.h

```diff
@@ -0,0 +1,52 @@
+#include "sm2.h"
+
+static inline int sm2_alloc_rma_ctx(struct sm2_ep *ep,
+				    struct sm2_sar_ctx **rma_ctx)
+{
+	struct sm2_sar_ctx *ctx;
+
+	ctx = ofi_buf_alloc(ep->sar_ctx_pool);
+	if (!ctx) {
+		FI_WARN(&sm2_prov, FI_LOG_EP_CTRL,
+			"Error allocating rma ctx\n");
+		return -FI_ENOMEM;
+	}
+	*rma_ctx = ctx;
+	return FI_SUCCESS;
+}
+
+static inline void sm2_free_rma_ctx(struct sm2_sar_ctx *rma_ctx)
+{
+	ofi_buf_free(rma_ctx);
+}
+
+void sm2_fill_rma_ctx(struct sm2_ep *ep, const struct fi_msg_rma *msg,
```

Comment:
nit: i sort of hate declarations between definitions, can we have all the declarations above or below the definitions?

Reply:
Resolved when I moved some functions into sm2_fifo.h